producers/mimosa26_sbg/src/M26Producer.cxx

```diff
@@ -0,0 +1,511 @@
+      try{
+	std::lock_guard<std::mutex> lck(m_mutex);
+	// check an read
+	std::cout << " reading" << std::endl;
+
+	// structure of this part depends on how data is send from SBG LV DAQ
+	// - likely to be package with 4 frames per trigger -> 4*n frames = 1 acquisition
+	// ->Is "4" configurable?
+	// - What of more than 1 trigger occurred during "4" frames
+	// -> Data dublication? Does the FW propagate trigger into into RAM such that we know which frames to dublicate
+	// -> Does the MAxTrigNumber influence the FW: longer BUSY? if not, the TLU does not know about this and will continue to issue triggers. 
+	//    Hence LVDAQ might not send data on a certain trigger, but all the other systems are! I think we either need a long busy ("4" frames) or we can not limit the number of triggers so we (need to) know how to send all matching frames to the producer and the producer can dublicate them in such a way that it sends the correct package of "4" frames to the Data Collector
+
+	//datalength1 = ni_control->DataTransportClientSocket_ReadLength("priv");
+	std::vector<unsigned char> mimosa_data_0(1);
+	//mimosa_data_0 = ni_control->DataTransportClientSocket_ReadData(datalength1);
+	unsigned char test = 125; 
+	mimosa_data_0.push_back(test);
+
+	//datalength2 = ni_control->DataTransportClientSocket_ReadLength("priv");
+	std::vector<unsigned char> mimosa_data_1(1);
+	//mimosa_data_1 = ni_control->DataTransportClientSocket_ReadData(datalength2);
+	test = 127; 
+	mimosa_data_1.push_back(test);
+
+
+	eudaq::RawDataEvent ev("NI", m_run, m_ev++);
+	ev.AddBlock(0, mimosa_data_0);
+	ev.AddBlock(1, mimosa_data_1);
+	SendEvent(ev);
+	eudaq::mSleep(283);
+	continue;
+
+      } catch (irc::DataNoEvent &) {
+	// no event in shrd mem
+	sched_yield();
+      }
+
+    }
+
+  } while (!m_terminated);
+}
+
+void M26Producer::OnConfigure(const eudaq::Configuration &param) {
+
+  m_config = param;
+
+  std::cout << "Configuring ...(" << m_config.Name() << ")" << std::endl;
+  try{
+    if (!m_configured) {
+
+      //m26_control->Connect(m_config);
+      std::cout << " Connecting to shared memory. " << std::endl;
+      try{
+
+        reset_com();
+
+        //one could also do: if(!IRC__FBegin()) but is less readible imho
+	ret = IRC__FBegin ( APP_VGErrUserLogLvl, APP_ERR_LOG_FILE, APP_VGMsgUserLogLvl, APP_MSG_LOG_FILE );
+	if(ret) throw irc::GlobalConnectError("Failed to perform global IRC init. Is LV running?");
+
+	ret = IRC_RCBT2628__FRcBegin ();
+	if(ret) throw irc::GlobalSharedMemError("Failed to find shared memory. Is LV running?");
+
+      } catch (const std::exception &e) { 
+	printf("while connecting: Caught exeption: %s\n", e.what());
+	//SetStatus(eudaq::Status::LVL_ERROR, "Stop error"); 
+	throw;
+      } catch (...) { 
+	printf("while connecting: Caught unknown exeption:\n");
+	//SetStatus(eudaq::Status::LVL_ERROR, "Stop error");
+	throw;
+      }
+
+      //m26_control->Init(m_config);
+      std::cout << " Initialising. " << std::endl;
+      try {

```

Comment:
probably just a style issue, but why do you wrap all of these calls in separate try-catch clauses?
if you fetch the mutex in one of them and the mainloop requests it, the loop will first be served before the next try-catch clause here will be executed. this might lead to problems since you just initialized the communication and then jump somewhere else.